src/overlays/actors/ovl_En_Fd_Fire/z_en_fd_fire.c

```diff
@@ -21,23 +28,216 @@ const ActorInit En_Fd_Fire_InitVars = {
+};
+
+static CollisionCheckInfoInit2 sColChkInit = { 0x01, 0x0000, 0x0000, 0x0000, 0xFF };
+
+static DamageTable sDamageTable = { {
+    0x00, 0x00, 0x00, 0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x04, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+} };
+
+void EnFdFire_UpdatePos(EnFdFire* this, Vec3f* targetPos) {
+    f32 dist;
+    f32 xDiff;
+    f32 yDiff;
+    f32 zDiff;
+    xDiff = targetPos->x - this->actor.posRot.pos.x;
+    yDiff = targetPos->y - this->actor.posRot.pos.y;
+    zDiff = targetPos->z - this->actor.posRot.pos.z;
+    dist = sqrtf(SQ(xDiff) + SQ(yDiff) + SQ(zDiff));
+    if (fabsf(dist) > fabsf(this->actor.speedXZ)) {
+        this->actor.velocity.x = (xDiff / dist) * this->actor.speedXZ;
+        this->actor.velocity.z = (zDiff / dist) * this->actor.speedXZ;
+    } else {
+        this->actor.velocity.x = 0.0f;
+        this->actor.velocity.z = 0.0f;
+    }
+
+    this->actor.velocity.y += this->actor.gravity;
+    if (!(this->actor.minVelocityY <= this->actor.velocity.y)) {
+        this->actor.velocity.y = this->actor.minVelocityY;
+    }
+}
+
+s32 EnFdFire_CheckCollider(EnFdFire* this, GlobalContext* globalCtx) {
+    if (this->actionFunc == EnFdFire_Disappear) {
+        return false;
+    }
+
+    if (this->collider.base.acFlags & 2) {
+        this->collider.base.acFlags &= ~2;
+        return true;
+    }
+
+    if (this->collider.base.maskB & 1) {
+        this->collider.base.maskB &= ~1;
+        return true;
+    }
+    return false;
+}
+
+void EnFdFire_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnFdFire* this = THIS;
+    s32 pad;
+    Player* player = PLAYER;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawFunc_Circle, 20.0f);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    func_80061EFC(&this->actor.colChkInfo, &sDamageTable, &sColChkInit);
+    this->actor.flags &= ~1;
+    this->actor.gravity = -0.6f;
+    this->actor.speedXZ = 5.0f;
+    this->actor.velocity.y = 12.0f;
+    this->spawnRadius = Math_Vec3f_DistXYZ(&this->actor.posRot.pos, &player->actor.posRot.pos);
+    this->scale = 3.0f;
+    this->tile2Y = (s16)Math_Rand_ZeroFloat(5.0f) - 0x19;
+    this->actionFunc = func_80A0E70C;
+}
+
+void EnFdFire_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnFdFire* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80A0E70C(EnFdFire* this, GlobalContext* globalCtx) {
+    Vec3f velocity = { 0.0f, 0.0f, 0.0f };
+    Vec3f targetPos = this->actor.parent->posRot.pos;
+
+    targetPos.x += this->spawnRadius * Math_Sins(this->actor.posRot.rot.y);
+    targetPos.z += this->spawnRadius * Math_Coss(this->actor.posRot.rot.y);
+    EnFdFire_UpdatePos(this, &targetPos);
+    if (this->actor.bgCheckFlags & 1 && (!(this->actor.velocity.y > 0.0f))) {
```

Comment:
```suggestion
    if (!(!(this->actor.bgCheckFlags & 1) || (this->actor.velocity.y > 0.0f)) {
```
is consistent with the check in EnFw